feat(vue): Rework Vue documentation to include Vue2 and Vue3 support (#3906)
getsentry/sentry-docs: src/includes/getting-started-config/javascript.vue.mdx

To initialize Sentry in your Vue application, add this to your `app.js`:

### Vue 2

```js
import Vue from "vue";
import * as Sentry from "@sentry/vue";
import { Integrations } from "@sentry/tracing";

Sentry.init({
  Vue,
  dsn: "___PUBLIC_DSN___",
  integrations: [
    new Integrations.BrowserTracing({

>       routingInstrumentation: Sentry.vueRouterInstrumentation(router),

Reviewer: Need to define `router` above like we do for Vue 3

    new Integrations.BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
      tracingOrigins: ["localhost", "my-site-url.com", /^\//],
    }),
  ],
  // Set tracesSampleRate to 1.0 to capture 100%
  // of transactions for performance monitoring.
  // We recommend adjusting this value in production
  tracesSampleRate: 1.0,
});

app.use(router);
app.mount("#app");
```

Additionally, Vue 3 allows you to use multiple apps with the same Sentry SDK instance, as well as add more apps dynamically after the SDK has been already initialized.

### Vue 3 - Multiple Apps

```js
const appOne = Vue.createApp(App);
const appTwo = Vue.createApp(App);
const appThree = Vue.createApp(App);

Sentry.init({
  app: [appOne, appTwo, appThree],
});
```

### Vue 3 - Manual Initialization

```js
import * as Sentry from "@sentry/vue";

// ...
const app = createApp(App);

Sentry.init({
  app,
  // ...
});

const miscApp = createApp(MiscApp);
miscApp.mixin(Sentry.createTracingMixins({ trackComponents: true }));
Sentry.attachErrorHandler(miscApp, { logErrors: true });
```

The SDK accepts a few different configuration options that let you change its behavior:

> - `attachProps` (defaults to `true`) - include all Vue components' props with the events

Reviewer 2: ```suggestion
- `attachProps` (defaults to `true`) - Includes all Vue components' props with the events.
```

> - `logErrors` (defaults to `false`) - decides whether SDK should call Vue's original `logError` function as well

Reviewer 2: ```suggestion
- `logErrors` (defaults to `false`) - Decides whether SDK should call Vue's original `logError` function as well.
```

> - `trackComponents` (defaults to `false`) - decides whether to track components by hooking into its lifecycle methods. Can be either set to `boolean` to enable/disable tracking for all of them. Or to an array of specific component names (case-sensitive).

Reviewer 2: ```suggestion
- `trackComponents` (defaults to `false`) - Decides whether to track components by hooking into its lifecycle methods. Can be set to either `boolean`, to enable/disable tracking for all of them, or to an array of specific component names (case-sensitive).
```

> - `timeout` (defaults to `2000`) - Time in miliseconds dictating how long to wait until the tracked root activity is marked as finished and sent of to Sentry.

Reviewer 2: ```suggestion
- `timeout` (defaults to `2000`) - Time in milliseconds dictating how long to wait until the tracked root activity is marked as finished and sent off to Sentry.
```
